packages/authentication/src/types.ts

```diff
@@ -35,6 +35,10 @@ export interface AuthenticateFn {
   (request: Request): Promise<UserProfile | undefined>;
 }
 
+export interface UserToUserProfileConverterFn<U> {
```

Comment:
Maybe name it as `UserProfileBuilder`?

Reply:
+1 to find a better name.

Personally, I would use `UserProfileFactory` (see [Factory pattern](https://en.wikipedia.org/wiki/Factory_method_pattern)).

The name `UserProfileBuilder` suggest a well-known [Builder pattern](https://en.wikipedia.org/wiki/Builder_pattern), which we are NOT following in this API.